### PR TITLE
Fix for #2458

### PR DESCRIPTION
**Pull request by briskycat**
A possible fix for https://github.com/tux3/qTox/issues/2458

### CODE REVIEW

File: src/widget/widget.cpp [hunk 1]
```diff
@@ -144,7 +144,10 @@ void Widget::init()
     actionLogout->setIcon(prepareIcon(":/img/others/logout-icon.svg", icon_size, icon_size));
 
     actionQuit = new QAction(this);
+#ifndef Q_OS_OSX
     actionQuit->setMenuRole(QAction::QuitRole);
+#endif
+    actionQuit->setShortcut(QKeySequence::Quit);
```
Comment (agilob) [inline src/widget/widget.cpp:150]: afair @antis81 had better idea of global quit button? @antis81 you ok with merging it?

Comment (antis81) [inline src/widget/widget.cpp:150]: Well, yes. The other one is marked WIP and doesn't work like expected yet. -> See #2524

**Comment by zetok** [comment]
> Merge pull request #1 from tux3/master

@briskycat that can be avoided by 

```
git fetch original && git rebase original/master master
```

 `:)`

**Comment by antis81** [comment]
@zetok Branches must be exactly vice versa. :)

I added a `main` (aka tux3) remote to my local repo, that allows me to do the following: 

``` bash
# set the master branch to follow the "tux3" main repository
git branch --set-upstream-to=main/master master

# always fetch (never pull) and update local master branch, that points to main/master
git fetch main master:master

# now, if I have an existing open PR branch, which is "behind master", I simply do: 
git rebase my-pr-branch master
```

Thus, I can follow master harmlessly and be always up-to-date. :)

**Comment by agilob** [comment]
Ready to merge?

**Comment by zetok** [comment]
@RowenStipe ↑ ?

**Comment by RowenStipe** [comment]
**Beep** **Boo** **Boop** It works.

**Comment by agilob** [comment]
Hmm. it adds ctrl+q to shourtcuts but when I have menu open and press ctrl+q nothing happens...

I'm not merging one fix to introduce another regression.

**Comment by RowenStipe** [comment]
@agilob The removal of line 150 

``` cpp
actionQuit->setShortcut(QKeySequence::Quit);
```

Builds and removes the shortcut cutting out the regression issue, at least in OSX.

**Comment by agilob** [comment]
@RowenStipe I know, I wanted @briskycat to fix it, but since I had to cherrypick anyway, it's merged.

Cherrypicked and pushed. https://github.com/tux3/qTox/commit/371900b56ff9389cff0cfabf909740bd2addfe0e

**Comment by zetok** [comment]
> when I have menu open and press ctrl+q

Wait, isn't that a X bug?

**Comment by agilob** [comment]
Is it?

**Comment by zetok** [comment]
I would think so - X cannot really into keyboard input with stuff like that, it will be fixed under wayland, but..

**Pull request closed without merging**